package/kernel/mac80211/patches/ath9k/551-ath9k_ubnt_uap_plus_hsr.patch

```diff
@@ -308,6 +309,8 @@
 +#ifndef HSR_H
 +#define HSR_H
 +
++#include "hw.h"
```

Comment:
another option is to just define the struct, like .e.g `mac.h` does to avoid requiring an include:

```suggestion
+struct ath_hw;
```

The header doesn't care what ath_hw looks like, just that it is something that can be pointed at.

Reply:
I reorganized the header files in `hsr.c` so that we can avoid including `hw.h` here. diff: https://github.com/openwrt/openwrt/compare/f434a28102960c43dc6225658007ba84e67b3792..1996a64c6e18bda6e1e3deffa1f742122eef8a16

Reply:
hsr.h uses `struct ath_hw`, so it should either include `hw.h` or define it itself, it shouldn't depend on users having included `hw.h` before.

This will then e.g. break if someone comes along and reorders the includes alphabetically (which isn't too uncommon).

Reply:
Thank you for your explanation. I added `struct ath_hw;` to the header file.